src/filesystem.cc

```diff
@@ -1322,10 +1347,15 @@ class S3FileSystem : public FileSystem {
   Status ParsePath(
       const std::string& path, std::string* bucket, std::string* object);
   Status CleanPath(const std::string& s3_path, std::string* clean_path);
-  Aws::SDKOptions options_;
-  s3::S3Client client_;
+  const Aws::SDKOptions options_;
+  s3::S3Client* client_;  // init after Aws::InitAPI is called
   re2::RE2 s3_regex_;
+  // enforce order: Aws::InitAPI -> Aws::ShutdownAPI -> Aws::InitAPI ...
```

Comment:
Elaborate on why

Reply:
Because AWS requires its S3 client library to be called exactly like: 
```
#include <aws/core/Aws.h>
int main(int argc, char** argv)
{
   Aws::SDKOptions options;
   Aws::InitAPI(options);
   {
      // make your SDK calls here.
   }
   Aws::ShutdownAPI(options);
   return 0;
}
```
[source](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/basic-use.html)

The `InitAPI` and `ShutdownAPI` are static. They should not be double called and should follow the order.